explainaboard/feature.py

```diff
@@ -278,350 +18,123 @@ class BucketInfo:
     method: str = "bucket_attribute_specified_bucket_value"
     number: int = 4
     setting: Any = 1  # For different bucket_methods, the settings are diverse
+    _type: Optional[str] = None
+
+    def __post_init__(self):
+        self._type: str = self.__class__.__name__
 
 
-# TODO(odashi): Make this class non-dataclass because __post_init__ does lots of
-# type-insensitive processes.
-@dataclass
-class ClassLabel:
-    """
-    (Most part of this class are from huggingface)
-    This class is used to define a new dataclass for sample's class label.
-    For example, in sentiment classification task, we have sample:
-        I love this movie \t  positive
-    in this case, "positive" is the class label.
-    There are three ways to create a ClassLabel object,
-    based on three objects:
-        * `num_classes`: create 0 to (num_classes-1) labels
-        * `names`: a list of label names
-        * `names_file`:a file that consists of the list of labels.
+def is_dataclass_dict(obj):
+    if (
+        not isinstance(obj, dict)
+        or '_type' not in obj.keys()
+        or obj['_type'] not in FEATURETYPE_REGISTRY.keys()
+    ):
+        return False
+    else:
+        return True
+
+
+def _fromdict_inner(obj):
+    # reconstruct the dataclass using the type tag
+    if is_dataclass_dict(obj):
+        result = {}
+        for name, data in obj.items():
+            result[name] = _fromdict_inner(data)
+        return FEATURETYPE_REGISTRY[obj["_type"]](**result)
+
+    # exactly the same as before (without the tuple clause)
+    elif isinstance(obj, (list, tuple)):
+        return type(obj)(_fromdict_inner(v) for v in obj)
+    elif isinstance(obj, dict):
+        return type(obj)(
+            (_fromdict_inner(k), _fromdict_inner(v)) for k, v in obj.items()
+        )
+    else:
+        return copy.deepcopy(obj)
 
-    Args:
-        num_classes: int, number of classes. If users adopt this way to create
-            an object, then all labels are str(numbers) less than num_classes.
-        names: list<str>
-        names_file:str, path to a file with names, one per line
-    """
 
-    num_classes: Optional[int] = None
-    names: Optional[list[str]] = None
+@dataclass
+class FeatureType:
```

Comment:
It'd also be good to have this documented, as people who are implementing new processors will have to know what the variables mean.